src/server/scripts/Northrend/zone_borean_tundra.cpp

```diff
@@ -94,183 +95,137 @@ struct npc_corastrasza : public ScriptedAI
-                if (Unit* temp = summon->GetSummonerUnit())
-                    if (Player* player = temp->ToPlayer())
-                        player->KilledMonsterCredit(me->GetEntry());
+        Player* playerCaster = caster->ToPlayer();
+        if (!playerCaster)
+            return;
 
-        if (GameObject* go_caribou = ObjectAccessor::GetGameObject(*me, go_caribouGUID))
-            go_caribou->SetGoState(GO_STATE_READY);
+        if (spellInfo->Id == SPELL_PLACE_FAKE_FUR)
+        {
+            _playerGUID = caster->GetGUID();
+            _placeFir = true;
+            _events.ScheduleEvent(EVENT_FUR_SPAWN, 1s);
+        }
     }
 
     void UpdateAI(uint32 diff) override
     {
-        if (phaseTimer <= diff)
+        if (!_placeFir)
+            return;
+
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
         {
-            switch (phase)
+            switch (eventId)
             {
-                case 1:
-                    me->SetVisible(true);
-                    phaseTimer = 2000;
-                    phase = 2;
-                    break;
-                case 2:
-                    if (GameObject* go_fur = me->FindNearestGameObject(GO_HIGH_QUALITY_FUR, 11.0f))
-                        me->GetMotionMaster()->MovePoint(0, go_fur->GetPositionX(), go_fur->GetPositionY(), go_fur->GetPositionZ());
-                    phaseTimer = 1500;
-                    phase = 3;
-                    break;
-                case 3:
-                    Talk(SAY_NESINGWARY_1);
-                    phaseTimer = 2000;
-                    phase = 4;
-                    break;
-                case 4:
-                    me->HandleEmoteCommand(EMOTE_ONESHOT_LOOT);
-                    phaseTimer = 1000;
-                    phase = 5;
-                    break;
-                case 5:
-                    me->HandleEmoteCommand(EMOTE_ONESHOT_NONE);
-                    phaseTimer = 500;
-                    phase = 6;
-                    break;
-                case 6:
-                    if (GameObject* go_fur = me->FindNearestGameObject(GO_HIGH_QUALITY_FUR, 11.0f))
-                        go_fur->Delete();
-                    phaseTimer = 500;
-                    phase = 7;
-                    break;
-                case 7:
+            case EVENT_FUR_SPAWN:
+                if (GameObject* fur = me->SummonGameObject(GO_HIGH_QUALITY_FUR, me->GetPosition(), QuaternionData(0.0f, 0.0f, 0.77162457f, 0.63607824f), 20s))
+                    _goFurGUID = fur->GetGUID();
+                _events.ScheduleEvent(EVENT_SPAWN_TRAPPER, 1s);
+                break;
+            case EVENT_SPAWN_TRAPPER:
+                if (TempSummon* trapper = me->SummonCreature(NPC_NESINGWARY_TRAPPER, me->GetPositionX() + (std::cos(me->GetOrientation()) * 21), me->GetPositionY() + (std::sin(me->GetOrientation()) * 21), me->GetPositionZ(), me->GetOrientation(), TEMPSUMMON_DEAD_DESPAWN, 6s))
```

Comment:
What is the intent behind this calculation? Any nearby position? Wouldn't using me->GetRandomPoint be better here? (and clearer)

Reply:
Its not random. The trapper spawns 21 yards away based on the orientation of the trap as I remember.

Reply:
In that case me->GetFirstCollisionPosition

Reply:
Done and tested. If you want the movepoint changed you will have to supply the code.